interfaces/cython/cantera/cti2yaml.py

```diff
@@ -0,0 +1,1648 @@
+        self._species1, self._species2 = species.split(' ')
+        self._acoeff = a_coeff
+        self._bcoeff = b_coeff
+
+
+class RedlichKwongMFTP(phase):
+    """A multi-component fluid model for non-ideal gas fluids.
+        """
+
+    def __init__(self, name='', elements='', species='', note='',
+                 reactions='none', kinetics='GasKinetics', initial_state=None,
+                 activity_coefficients=None, transport='None', options=[]):
+
+        phase.__init__(self,name, 3, elements, species, note, reactions,
+                       initial_state,options)
+        self.thermo_model = 'Redlich-Kwong'
+        self._kin = kinetics
+        self._tr = None if transport == 'None' else transport
+        self._activityCoefficients = activity_coefficients
+
+    def get_yaml(self, out):
+        super(RedlichKwongMFTP, self).get_yaml(out)
+        for section, names in self._sp:
+            if section != 'species':
+                _printerr("WARNING: Converting Redlich-Kwong species from"
+                    " different input files ({}) is not supported.".format(section))
+
+        spdict = {sp._name: sp for sp in _species}
+        for params in self._activityCoefficients:
+            if isinstance(params, pureFluidParameters):
+                sp = spdict[params._species]
+                sp.rk_pure = {'a': params._acoeff, 'b': params._bcoeff}
+            elif isinstance(params, crossFluidParameters):
+                sp1 = spdict[params._species1]
+                sp1.rk_binary[params._species2] = params._acoeff
+                sp2 = spdict[params._species2]
+                sp2.rk_binary[params._species1] = params._acoeff
+
+
+class constantIncompressible(object):
+    """Constant molar volume."""
+    def __init__(self, molarVolume=0.0):
+        """
+        :param molarVolume:
+            Reference-state molar volume. Default: 0.0.
+        """
+        self._mv = molarVolume
+
+
+class IdealSolidSolution(phase):
+    """An IdealSolidSolution phase."""
+    def __init__(self, name='', elements='', species='', note='',
+                 transport='None', initial_state=None,
+                 standard_concentration=None, options=[]):
+        phase.__init__(self, name, 3, elements, species, note, 'none',
+                       initial_state, options)
+        self.thermo_model = 'ideal-condensed'
+        self._stdconc = standard_concentration
+        if self._stdconc is None:
+            raise InputError('In phase {}: standard_concentration must be specified.', name)
+        self._tr = None if transport == 'None' else transport
+
+    def get_yaml(self, out):
+        super(IdealSolidSolution, self).get_yaml(out)
+        out['standard-concentration-basis'] = _newNames[self._stdconc]
+
+
+class table(thermo):
+    """User provided thermo table for BinarySolutionTabulatedThermo"""
+    def __init__(self, moleFraction=([],''), enthalpy=([],''), entropy=([],'')):
+        """
+        :param moleFraction:
```

Comment:
If these are all required, why do they have default values? Why is this class defined here and not with the rest of the `thermo` subclasses?